06/tasks/task_33_valid_parentheses.py

```diff
@@ -4,4 +4,20 @@ def valid_parentheses(s: str) -> bool:
     """
     Zjistí, zda má řetězec validní závorky.
     """
-
+    r_par = '()'  # round brackets
+    b_par = '[]'  # box brackets
+    c_par = '{}'  # curly bracket
+    while r_par in s or b_par in s or c_par in s:
+        if r_par in s:
+            pos = s.index(r_par)
+            s = s[:pos] + s[pos + 2:]
+        elif b_par in s:
+            pos = s.index(b_par)
+            s = s[:pos] + s[pos + 2:]
+        else:
+            pos = s.index(c_par)
+            s = s[:pos] + s[pos + 2:]
+    if s == '':
+        return True
+    else:
+        return False
```

Comment:
Je tam opakující se kód. Ta úloha redukování závorek by se dala řešit funkcí.

Dále se nabízí zkrácení:
```suggestion
    return s == ''
```
popř. `return not s`, ale to možná není tak čitelné.

Reply:
Ad opakující se kód - původně jsem tam funkci měla, ale to bych si pak musela přepsat test file, aby načítal i tu mou další funkci, ne?